crates/beautifier/src/beautifier.rs

```diff
@@ -253,7 +271,52 @@ impl<W: io::Write> Beautifier<W> {
     }
 
     fn beautify_new(&mut self, new: &ast::New<'_>, indent: usize) -> io::Result<()> {
+        if self.try_hopon {
+            if let Some(hop_on) = try_hopon(new) {
+                return self.beautify_simple(&hop_on, indent);
+            }
+        }
+        // else
         compound!(self, indent, new);
         Ok(())
     }
 }
+
+struct HopOn<'i> {
+    pub peer_id: ast::ResolvableToPeerIdVariable<'i>,
+}
+
+impl Display for HopOn<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "hopon {}", self.peer_id)
+    }
+}
+
+fn try_hopon<'i>(root_new: &ast::New<'i>) -> Option<HopOn<'i>> {
```

Comment:
Let's make a comment here re what is this pattern about and how does this pattern look in AIR code